### PR TITLE
Doc 324 redesign datastores draft

### PR DESCRIPTION
**Pull request by oojo12**
# What does this PR do?

Redesign datastores documentation to make content more accessible to users.

### CODE REVIEW

**Comment by oojo12** [comment]
@essiequoi for our meeting today. If you generally approve of the idea this is what it would look like.

**Comment by oojo12** [comment]
note - waiting on feedback on whether the Operationalize page should be broken up into separate pages with the following specifications:

1. Creating Datastores
2. Attaching Datastores
3. Periodically updating Datastores

**Comment by essiequoi** [comment]
@oojo12 let's break it down, like stated above. and i'm not too hot on "operationalizing datastores" as it's a little too buzzy. what about something else like "using datatstores" or "working with datastores"?

**Comment by oojo12** [comment]
sure

**Comment by oojo12** [comment]
@essiequoi addressed if there are any other requested changes feel free to commit directly to the branch.

**Comment by oojo12** [comment]
@rlizzo please review for technical correctness and add any technical content you think is missing.

**Comment by oojo12** [comment]
as a note many links won't work until the branch is merged. So I'd just go in and verify that the links are correct @essiequoi

File: docs/features/datastores/attaching-datastores.md
Comment (essiequoi) [inline docs/features/datastores/attaching-datastores.md:12]: we should explain that the 1 means datastore version

**Review by essiequoi — APPROVED** [review]

File: docs/features/datastores/overview.md
Comment (essiequoi) [inline docs/features/datastores/overview.md:8]: So there are filetypes and then there are upload methods. We should point out both - so for example HTTP URL to compressed directories tar.gz and zip folders

File: docs/features/datastores/overview.md
Comment (essiequoi) [inline docs/features/datastores/overview.md:11]: Make training iteration quicker, because no need to download the data for your script every time. And data is local to run/session machine

File: docs/features/datastores/overview.md
Comment (essiequoi) [inline docs/features/datastores/overview.md:15]: Unclear if we will support private s3 for grid cloud

**Review by essiequoi — CHANGES_REQUESTED** [review]

File: docs/features/datastores/attaching-datastores.md
Comment (oojo12) [reply docs/features/datastores/attaching-datastores.md:12]: that "1" should be removed. I learned late last week that mounting datastores on sessions places them at `/datastores/<datastore-name>` if memory serves correct

File: docs/features/datastores/overview.md
Comment (oojo12) [reply docs/features/datastores/overview.md:11]: not exactly true you still have to transfer the data from S3 to the Session or Run which makes Datastores worse than having the data on local disk. We need to be careful here.

File: docs/features/datastores/overview.md
Comment (oojo12) [reply docs/features/datastores/overview.md:15]: It is something that was said should work and is something that there is a linear ticket for. Whether it is a documentation thing or a broken feature is something that needs to be discerned.

File: docs/features/datastores/overview.md
Comment (oojo12) [reply docs/features/datastores/overview.md:8]: All I know for upload methods are http and s3 buckets. Feel free to update with your own commit addressing this request as there may be more.

File: docs/features/datastores/attaching-datastores.md
Comment (oojo12) [reply docs/features/datastores/attaching-datastores.md:12]: confirmation screenshot. That behavior change was never communicated or it is a regression.
Screenshot 2022-03-27 222858

File: docs/features/datastores/overview.md
Comment (oojo12) [reply docs/features/datastores/overview.md:11]: The only obvious benefit is when the data is too large for your local machine. But even then you can lazily load it sometimes so we should be very intentional and thorough here. @rlizzo input will be important for the community.

Comment (essiequoi) [reply docs/features/datastores/overview.md:11]: we're saying the same thing, i think.. at least about the datastore access - no need to download it from your script on every iteration. but not sure i fully agree with the caching - do we have that yet? @rlizzo

**Comment by oojo12** [comment]
closing to make against dev instead.

**Pull request closed without merging**
